### ISSUE TITLE
Dynastic Cycle 'push towards/away from Expansion' catalysts apply to wrong era during Expansion

### ISSUE THREAD

**Issue by pharaox** [ProZeratul/CrusaderKings3UnofficialPatch#551]
**Reported on Steam by `Nider001` on `2026-06-26 14:26`** (https://steamcommunity.com/sharedfiles/filedetails/?id=2871648329).

> The "House Head consulted Heaven — push towards Expansion" and "Movement consulted Heaven — push towards Expansion" Dynastic Cycle catalysts contribute towards Tension era while the current era is Expansion. Meanwhile, both respective "... — push away from Expansion" catalysts contribute away from Tension era. This setup just doesn't make sense - since the current era is Expansion, the catalyst effects should be reversed. I didn't check if the same happens during Advancement era, but it's worth investigating as well.

**Comment by Nider001** [comment]
To clarify, by writing "I didn't check if the same happens during Advancement era, but it's worth investigating as well" I meant that the "... — push towards Advancement" catalysts should be checked too, to see if they need to be reversed during Advancement era in a similar manner.

**Comment by pharaox** [comment]
**analyzer:** Confirmed.

- In `tgp_dynastic_cycle.txt`, the **`stability_expansion → instability`** future-phase catalyst block (≈ lines 715–720) sets the "push towards Expansion" catalysts (`..._expansion_positive`) to `..._gain` toward Instability and the "push away from Expansion" ones (`..._expansion_negative`) to `..._loss`. Per the loc (`_expansion_positive_desc` = "push towards Expansion", `_negative` = "push away"), this is **reversed**: while already in Expansion, "towards Expansion" should resist leaving (loss), "away" should accelerate it (gain).
- The same `_expansion_*` catalysts are used correctly elsewhere — in the *instability → stability_expansion* blocks `_positive = gain` correctly accelerates *entering* Expansion. The bug is isolated to the Expansion phase's own decay block.
- **Advancement era**: no analogous bug. The `stability_advancement → instability` block uses `consulted_heaven_instability_*` (not `advancement_*`) catalysts with correct polarity. The Expansion phase is the outlier that both uses the era-specific catalysts in its decay block and reverses their polarity.

Vanilla; the file isn't overridden by Unop.

**Classification:** `confirmed` / `vanilla` / `common` — confidence `high`

**Comment by pharaox** [comment]
<details><summary>🤖 handover document</summary>

---
issue: 551
title: "Dynastic Cycle 'push towards/away from Expansion' catalysts apply to wrong era during Expansion"
status: confirmed
origin: vanilla
area: [common]
confidence: high
---

## Analysis

**Reported:** During the **Expansion** stability phase, the "consulted Heaven — push towards Expansion" catalysts (house head & movement) push the Dynastic Cycle *towards* the Tension/Instability era, while the "push away from Expansion" catalysts push *away* from it. Since the current era is already Expansion, the effects are reversed. Reporter also asks whether Advancement era has the analogous problem.

**Files checked:**
- `${CK3_BASE_DIR}/common/situation/situations/tgp_dynastic_cycle.txt` (vanilla, **not** overridden by Unop) — the situation phase/`future_phases`/`catalysts` definitions.
- `${CK3_BASE_DIR}/common/situation/catalysts/10_tgp_dynastic_cycle_catalysts.txt` — catalyst declarations.
- `${CK3_BASE_DIR}/common/scripted_effects/10_dlc_tgp_dynastic_cycle_scripted_effects.txt` — `tgp_catalyst_from_decision_effect`, which maps a movement consulting Heaven to a catalyst.
- `${CK3_BASE_DIR}/localization/english/dlc/tgp/dlc_tgp_situation_catalysts_l_english.yml` — confirms `_expansion_positive_desc` = "push **towards** Expansion", `_expansion_negative_desc` = "push **away from** Expansion".

**Findings:**
- Each phase's `future_phases → <target> → catalysts` block lists how each catalyst moves progress **towards that target phase** (`..._gain` = progress toward target; `..._loss` = progress away).
- In phase **`situation_dynastic_cycle_phase_stability_expansion`**, the `future_phases → situation_dynastic_cycle_phase_instability` block (lines ~715–720) sets:
  - `catalyst_movement_consulted_heaven_expansion_positive = massive_situation_catalyst_gain`
  - `catalyst_movement_consulted_heaven_expansion_negative = massive_plus_situation_catalyst_loss`
  - `catalyst_house_head_consulted_heaven_expansion_positive = major_situation_catalyst_gain`
  - `catalyst_house_head_consulted_heaven_expansion_negative = major_plus_situation_catalyst_loss`
- Per the loc, `_expansion_positive` = "push **towards** Expansion" yet it adds **gain toward Instability** (i.e. accelerates *leaving* Expansion); `_expansion_negative` = "push **away from** Expansion" yet it **resists** leaving. Root cause: **reversed polarity** for the expansion catalysts while the current phase is Expansion.
- The catalysts themselves are correct elsewhere: in the *instability → stability_expansion* future-phase blocks (lines ~1298–1301 and ~1606–1609) `_expansion_positive = gain` correctly accelerates entry **into** Expansion. The bug is specific to the Expansion phase's own decay-to-Instability block.
- **Advancement era**: the `stability_advancement → instability` block (lines ~983–1190) does **not** use the `advancement_*` consulted-Heaven catalysts at all — it uses `consulted_heaven_instability_positive/negative` (push toward/away instability), which have correct polarity. So Advancement era does **not** show the reversed-polarity bug. The expansion phase is the outlier that (a) uses the era-specific `expansion_*` catalysts in its decay block and (b) does so with reversed polarity.
- Reachability: reproduces in current vanilla (1.19); not overridden by Unop → `vanilla`.

**Pattern class:** Dynastic Cycle `future_phases` catalyst polarity. Members: every (current phase → target phase) catalyst block.
- *instability → stability_expansion / stability_advancement*: era catalysts used with `positive = gain` — correct (push towards a phase accelerates entering it).
- *stability_advancement → instability*: uses `instability_*` catalysts — correct.
- *stability_expansion → instability*: uses `expansion_*` catalysts with `positive = gain` — **incorrect**, because here "towards Expansion" should *resist* the transition out of Expansion, not accelerate it. This is the lone outlier.

**Fix directions (soft):**
- Option A (minimal, matches reporter): in the `stability_expansion → instability` block, **swap the gain/loss** on the four expansion catalysts so `_expansion_positive` resists the decay (`..._loss`) and `_expansion_negative` accelerates it (`..._gain`).
- Option B (mirror the advancement phase): replace the `expansion_*` catalysts in that block with `instability_*` catalysts, making the Expansion decay block symmetric with the Advancement decay block — though that changes which player action drives the timer, so it's a larger design change.
- Option A is the more localized, lower-risk fix and directly reflects the loc semantics.

**Classification:** `confirmed` / `vanilla` / `common` — confidence `high`


## Fix 1

**Approach:** In the `stability_expansion → instability` future-phase block, flip the direction keyword (`gain`↔`loss`) on the four `..._consulted_heaven_expansion_positive/negative` catalysts, keeping each line's magnitude tier — so "towards Expansion" resists decay (`_loss`) and "away from Expansion" accelerates it (`_gain`), mirroring the correctly-polarized `_instability_*` siblings. (Option A from the analysis; rejected Option B as a larger design change.)

**Files changed:**
- `common/situation/situations/tgp_dynastic_cycle.txt` — four lines (715/716/719/720) in the expansion-phase decay block; edited by line position because the *correct* instability→expansion blocks (1298-1301, 1606-1609) contain byte-identical text and must not be touched. File added unmodified in a prior commit.
- `ck3-tiger.conf` — Ignored two pre-existing, unrelated vanilla scope warnings surfaced by adding the file (see below).

**Tiger (via tiger skill):** Adding the situation file surfaced two `warning(scopes): root produces situation but expected character` at lines 1214 and 1531 — both `trigger_situation_catalyst = { character = root }` inside phase `on_start` blocks. Per `_situations.info`, a phase `on_start`'s `Root = situation`, so passing `root` as `character` is a genuine **vanilla** scope bug, unrelated to the polarity issue. Classified **Ignored** with a precise text+file rule (grouped next to the frankokratia-faith ignore). **Follow-up flagged** for a separate issue (intended character is likely the hegemon / last China holder).

**Key assumption:** `gain` adds progress toward the target phase (Instability) and `loss` subtracts (confirmed by `10_tgp_dynastic_cycle_values.txt`: `_loss = _gain * -1`); the loc confirms `_expansion_positive` = "push towards Expansion" / `_expansion_negative` = "push away". So during Expansion, "towards" must resist (loss) and "away" must accelerate (gain). The instability siblings in the same block (positive=gain, negative=loss) being correct corroborates the intended pattern. In-game: during Expansion, consult-Heaven "push towards Expansion" should now lengthen the Expansion era; "push away" should shorten it.

**Commit:** Fix reversed Dynastic Cycle expansion catalyst polarity during Expansion phase (+ Add tgp_dynastic_cycle.txt unmodified)

**PR:** https://github.com/ProZeratul/CrusaderKings3UnofficialPatch/pull/574



## Review 1

**Verdict:** Looks good

Verified the polarity swap, its targeting, and the tiger ignore:

- **Right block, surgically targeted.** The four edited lines (715/716/719/720) are confirmed inside the `stability_expansion` phase (def line 628) → `future_phases → situation_dynastic_cycle_phase_instability` decay block — the lone outlier from the analysis. `diff` vs vanilla shows *exactly* those four lines changed; the byte-identical sibling blocks (instability→expansion at 1298-1301 and 1606-1609, correct as-is) are untouched. The line-position editing concern was handled correctly.
- **Swap is a pure sign flip.** Confirmed in `10_tgp_dynastic_cycle_values.txt` that `_loss = _gain × -1`, so each line keeps its magnitude tier and only flips sign: `expansion_positive` massive_gain(+50)→massive_loss(−50) now *resists* decay ("towards Expansion"); `expansion_negative` massive_plus_loss(−75)→massive_plus_gain(+75) now *accelerates* it ("away"). Matches the loc semantics exactly. The `instability_*` siblings (positive=gain) left untouched corroborate the intended pattern.
- **Advancement era correctly excluded.** Its decay block uses `instability_*` catalysts (correct polarity), not `advancement_*` — so no analogous bug, as the analysis concluded.
- **Tiger ignore justified, independently verified.** The two `root produces situation but expected character` warnings (lines 1214, 1531) are `trigger_situation_catalyst = { character = root }` inside phase `on_start` blocks where `_situations.info` confirms `Root = situation` — a genuine pre-existing vanilla scope bug, unrelated to the polarity fix, requiring its own analysis (intended scope is likely the hegemon). Tiger is *correct*, so `# Ignored` is the right section; precise text+file rule, grouped with the frankokratia scope ignore. Follow-up flagged.
- **Mechanics.** Vanilla file added byte-identical in its own commit; fix in a follow-up commit. `#Unop` comments on all changed lines, accurate. `make tiger` re-run by reviewer: clean.

No High/Medium findings.

**Findings:**

| Severity | Description |
|---|---|
| Low | The deferred `trigger_situation_catalyst { character = root }` scope bug (situation passed where character expected, phase `on_start`) is real and worth its own issue, as flagged — not blocking. |



## Fix 2

**Approach:** Fix (rather than ignore) the pre-existing vanilla scope bug the maintainer asked about: in the `instability_conquest` and `instability` phase `on_start` blocks, `trigger_situation_catalyst = { catalyst = catalyst_hegemon_lost_mandate_of_heaven; character = root }` passed `root` (which is the *situation* in a phase `on_start`, per `_situations.info`) where a *character* is expected. Comment out the invalid `character = root` so the catalyst still fires/records via the no-attribution form, and remove the now-unneeded `ck3-tiger.conf` scope ignore.

**Addresses review:** Resolves the maintainer `@fixer` request on PR #574 / issue #551 ("Fix the tiger reported issues, instead of ignoring it") — the two `root produces situation but expected character` warnings (lines 1214/1531) are now fixed, not ignored. This supersedes the Review 1 Low finding, which had merely flagged the deferred scope bug as worth its own issue.

**Files changed:**
- `common/situation/situations/tgp_dynastic_cycle.txt` — comment out the invalid `character = root` in both lost-mandate catalyst blocks (lines 1214, 1531), edited by line number because the two blocks are byte-identical. The Fix 1 polarity edit is unchanged.
- `ck3-tiger.conf` — remove the scope ignore added in Fix 1.

**Key assumption:** `trigger_situation_catalyst`'s `character` argument is optional and used only for catalyst-history attribution — corroborated by (a) the epidemic catalysts a few lines above in the same file using the short `trigger_situation_catalyst = catalyst_X` form with no character, and (b) `_catalysts.info` defining no per-catalyst schema. Since these blocks only run when `NOT exists title:h_china.holder` (no current hegemon character) and `root` was an invalid scope, dropping the arg removes a bogus attribution without changing the catalyst's gameplay effect. In-game: during a lost-mandate transition (instability_conquest / instability phase entry with China holderless), the catalyst should still register in the Dynastic Cycle (no error.log scope error).

**Commit:** Drop invalid character arg on lost-mandate catalyst in phase on_start

**PR:** https://github.com/ProZeratul/CrusaderKings3UnofficialPatch/pull/574 (existing PR updated)
</details>

**Comment by pharaox** [comment]
`@fixer` Fix the tiger reported issues, instead of ignoring it.
